Chatbot.py

```diff
@@ -428,6 +428,40 @@ async def runAgent(userQuery, chat_history):
     # iconIndex = index % len(listIcons)
     questionBtns.append(st.button(f":large_green_circle: {question}", type="secondary"))
 
+if 'clicked' not in st.session_state:
+    st.session_state.clicked = False
+clickedQuestion = None
+
+if 'clicked_question' not in st.session_state:
+    st.session_state.clicked_question = None
+
+def click_button(question):
+    print("button clicked!" + question)
+    st.session_state.clicked = True
+    st.session_state.clicked_question = question
+
+follow_up_questions = []
+followUpBtns = []
+def suggestFollowUps():
+    follow_up_query = f"""
+    Given this chat history {chat_history}, Suggest 2 follow-up questions the user 
+    might ask next. Give your answer in the format where each line has a separate 
+    question. Do not number your responses."""
```

Comment:
I left this comment earlier but github may have dropped it: consider using OpenAI's tool calling features for something like this, which lets you coerce the response to a desired schema and avoid prompting this way / parsing out the list later.

Example (using LangChain):
```python
from typing import List

from langchain_core.pydantic_v1 import BaseModel
from langchain_openai import ChatOpenAI


class FollowUpQuestions(BaseModel):
    """Follow up questions."""
    questions: List[str]

llm = ChatOpenAI(model="gpt-4o-2024-05-13", temperature=0)
structured_llm = llm.with_structured_output(FollowUpQuestions)
response = structured_llm.invoke(follow_up_query)
response.questions  # <-- will be a list of strings
```
Under the hood this just wraps OpenAI's tool calling features, which you can also access via their sdk: https://platform.openai.com/docs/guides/function-calling